http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -112,6 +114,7 @@ public NettyRequestExecutor(RequestContext context) {
                 if (!channelPromise.isSuccess()) {
                     return;
                 }
+                log.error("Exception caught but promise was already done", t);
```

Comment:
`warn` maybe?

Reply:
Actually I'll just remove this for now, I don't think this log line is necessary. This happens when the execution future is cancelled externally.

Reply:
@dagnir @zoewangg - this was super important when troubleshooting my changes in #1478. I would strongly suggest keeping a log statement here at either `error` or `warn` level.

I was surprised by how often that block of code got hit during the automated tests.